es-space-mgmt/main/default/lwc/reservationDetailForm/__tests__/reservationDetailForm.test.js

```diff
@@ -0,0 +1,244 @@
+import { createElement } from 'lwc';
+import getMarketsByState from '@salesforce/apex/marketServices.getMarketsByState';
+import { registerApexTestWireAdapter } from '@salesforce/sfdx-lwc-jest';
+import ReservationDetailForm from 'c/reservationDetailForm';
+
+// Register as Apex wire adapter. Some tests verify that provisioned values trigger desired behavior.
+const getMarketsByStateAdapter = registerApexTestWireAdapter(getMarketsByState);
+
+// Realistic data with a list of markets
+const mockMarketsByStateRecords = require('./data/getMarketsByState.json');
+
+// Realistic data with a list of markets returning 0 rows
+const mockMarketsByStateNoRecords = require('./data/getMarketsByStateNoRecords.json');
+
+// Constants for city, state and Start
+const CITY = 'San Francisco';
+const STATE = 'CA';
+const START_DATE = '1/1/2020';
+
+const CITY_OPTIONS = [
+    { value: '0031700000pJRRSAA4', label: 'San Francisco' },
+    { value: '0031700000pJRRSAA5', label: 'San Ramon' }
+];
+
+describe('c-reservation-detail-form', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    it('shows error panel element', () => {
+        const element = createElement('c-reservation-detail-form', {
+            is: ReservationDetailForm
+        });
+        document.body.appendChild(element);
+        // Emit error from @wire
+        getMarketsByStateAdapter.error();
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            const errorPanelEl = element.shadowRoot.querySelector(
+                'c-error-panel'
+            );
+            expect(errorPanelEl).not.toBeNull();
+            expect(errorPanelEl.friendlyMessage).toBe(
+                'There was an issue loading related market data.'
+            );
+        });
+    });
+
+    it('getMarketsByState @wire data', () => {
```

Comment:
This test does two things, so it should be splitted.

Test 1: Check if values that are passed down via `@api` are showing up the template elements as expected.
Test 2: Check wire adapter.

Reply:
I have split these now as per suggestion!